korge/src/jvmMain/kotlin/com/soywiz/korge/testing/testing_utils.kt

```diff
@@ -0,0 +1,339 @@
+package com.soywiz.korge.testing
+
+import com.soywiz.klock.DateFormat
+import com.soywiz.klock.DateTime
+import com.soywiz.korge.Korge
+import com.soywiz.korge.annotations.KorgeExperimental
+import com.soywiz.korge.input.onClick
+import com.soywiz.korge.testing.KorgeTesterUtils.makeGoldenFileNameWithExtension
+import com.soywiz.korge.ui.uiButton
+import com.soywiz.korge.ui.uiScrollable
+import com.soywiz.korge.view.Container
+import com.soywiz.korge.view.ScalingOption
+import com.soywiz.korge.view.Stage
+import com.soywiz.korge.view.View
+import com.soywiz.korge.view.Views
+import com.soywiz.korge.view.alignLeftToRightOf
+import com.soywiz.korge.view.alignTopToBottomOf
+import com.soywiz.korge.view.centerOn
+import com.soywiz.korge.view.centerXOn
+import com.soywiz.korge.view.centerXOnStage
+import com.soywiz.korge.view.container
+import com.soywiz.korge.view.image
+import com.soywiz.korge.view.renderToBitmap
+import com.soywiz.korge.view.scaleWhileMaintainingAspect
+import com.soywiz.korge.view.solidRect
+import com.soywiz.korge.view.text
+import com.soywiz.korim.bitmap.Bitmap
+import com.soywiz.korim.color.Colors
+import com.soywiz.korim.format.PNG
+import com.soywiz.korim.format.readBitmap
+import com.soywiz.korim.format.writeBitmap
+import com.soywiz.korio.async.suspendTest
+import com.soywiz.korio.file.VfsFile
+import com.soywiz.korio.file.std.localCurrentDirVfs
+import com.soywiz.korma.geom.ISizeInt
+import kotlinx.coroutines.sync.Mutex
+import java.awt.HeadlessException
+
+object KorgeTesterUtils {
+    fun makeGoldenFileNameWithExtension(testMethodName: String, goldenName: String) =
+        "${testMethodName}_$goldenName.png"
+}
+
+sealed class KorgeTestResult(
+    open val testMethodName: String,
+    open val goldenName: String
+) {
+    abstract suspend fun acceptChange()
+
+    val goldenFileNameWithExt: String
+        get() = makeGoldenFileNameWithExtension(testMethodName, goldenName)
+
+    data class Diff(
+        override val testMethodName: String,
+        override val goldenName: String,
+        val testGoldensVfs: VfsFile,
+        val tempVfs: VfsFile,
+        val oldBitmap: Bitmap,
+        val newBitmap: Bitmap
+    ) : KorgeTestResult(testMethodName, goldenName) {
+        override suspend fun acceptChange() {
+            tempVfs[goldenFileNameWithExt].copyTo(testGoldensVfs[goldenFileNameWithExt])
+        }
+    }
+
+    data class Deleted(
+        override val testMethodName: String,
+        override val goldenName: String,
+        val testGoldensVfs: VfsFile,
+        val oldBitmap: Bitmap,
+    ) : KorgeTestResult(testMethodName, goldenName) {
+        override suspend fun acceptChange() {
+            println("Deleting golden! $goldenFileNameWithExt")
+            testGoldensVfs[goldenFileNameWithExt].delete()
+        }
+    }
+}
+
+data class KorgeTestResults(
+    val testMethodName: String,
+    val results: MutableList<KorgeTestResult> = mutableListOf()
+)
+
+class KorgeTester(
```

Comment:
Can we rename it to `KorgeScreenshotTester` as well as the `korgeTest` -> `korgeScreenshotTest`?

Reply:
Done